resources.tf

```diff
@@ -1,8 +1,168 @@
+#############  IAM - policy document for breakglass, CI, administrator
+# Create PDM role
+resource "aws_iam_role" "pdm" {
+  name               = "pdm"
+  assume_role_policy = data.aws_iam_policy_document.ec2_assume_role.json
+  tags               = local.tags
+}
+
+#Get resources
+data "aws_iam_user" "breakglass" {
+  user_name = "breakglass"
+}
+
+data "aws_iam_role" "ci" {
+  name = "ci"
+}
+
+data "aws_iam_role" "administrator" {
+  name = "administrator"
+}
+
+data "aws_iam_role" "aws_config" {
+  name = "aws_config"
+}
+
+data "aws_iam_role" "analytical_dataset_generator" {
+  name = "analytical_dataset_generator"
+}
+
+#Create policy document
+data "aws_iam_policy_document" "published_bucket_kms_key" {
+  
+  statement {
+    sid    = "EnableIAMPermissionsBreakglass"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_user.breakglass.arn]
+    }
+
+    actions   = ["kms:*"]
+    resources = ["*"]
+  }
+
+  statement {
+    sid       = "EnableIAMPermissionsCI"
+    effect    = "Allow"
+    actions   = ["kms:*"]
+    resources = ["*"]
+
+    principals {
+      identifiers = [data.aws_iam_role.ci.arn]
+      type        = "AWS"
+    }
+  }
+
+  statement {
+    sid    = "DenyCIEncryptDecrypt"
+    effect = "Deny"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_role.ci.arn]
+    }
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:ImportKeyMaterial",
+      "kms:ReEncryptFrom",
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "EnableIAMPermissionsAdministrator"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_role.administrator.arn]
+    }
+
+    actions = [
+      "kms:Describe*",
+      "kms:List*",
+      "kms:Get*",
+      "kms:Put*",
```

Comment:
Please remove these `Put` and `Create` permissions. I think I understand the problem you were having now.

Were you trying to create the key/key policy yourself? If so, that's probably why it was failing. We should just PR this, then once merged, the CI role will be able to create the policy as it has `*` permissions. We, as administrators will have the ability to *read* they key's configuration which is all that we need to be able to do.